### PR TITLE
Removing corn oil from deep fryer with beakers

### PR DESCRIPTION
**Pull request by ghost**
[tweak][tested]

## What this does
Allows scooping up corn oil from deep fryers and similar machinery, current code only allows filling with reagents.

## Why it's good
Source of corn oil, consistency.

## Changelog
:cl:
 * rscadd: Corn oil and similar reagents can now be scooped up with a beaker from deep fryers.

### CODE REVIEW

**Comment by Hinaichigo** [comment]
It'd be cool if the oil was really hot so you could splash it on people and scald them with it and such, at least if the fryer is powered, or something.

**Comment by Servalfang** [comment]
you can already do this with syringes

**Comment by Eneocho** [comment]
>you can already do this with syringes

Yeah, but consistency and all that, you can beaker out of any reagent container except the deep fryers.

File: code/modules/reagents/reagent_containers.dm [hunk 1]
```diff
@@ -234,13 +234,19 @@ var/list/LOGGED_SPLASH_REAGENTS = list(FUEL, THERMITE)
...
+			var/obj/structure/reagent_dispensers/S = target
+			if(S.can_transfer(src, user))
+				var/tx_amount = transfer_sub(target, src, S.amount_per_transfer_from_this, user)
+				if (tx_amount > 0)
+					to_chat(user, "<span class='notice'>You fill \the [src][src.is_full() ? " to the brim" : ""] with [tx_amount] units of the contents of \the [target].</span>")
+				return tx_amount
+		if(reagents && reagents.is_empty() && istype(target, /obj/machinery/cooking/deepfryer))
+			var/tx_amount = transfer_sub(target, src, reagents.maximum_volume, user)
 			if (tx_amount > 0)
-				to_chat(user, "<span class='notice'>You fill \the [src][src.is_full() ? " to the brim" : ""] with [tx_amount] units of the contents of \the [target].</span>")
```
Comment (Hinaichigo) [inline code/modules/reagents/reagent_containers.dm:243]: What about cases where there's less oil in the fryer than the remaining capacity of the container? Like say using a beaker to scoop from a fryer with only 40 u of corn oil left?

Comment (Kurfursten) [reply code/modules/reagents/reagent_containers.dm:243]: He's right, it's not necessarily to the brim

Comment (ghost) [reply code/modules/reagents/reagent_containers.dm:243]: Fixed

**Pull request merged**